distributed/nanny.py

```diff
@@ -294,6 +302,13 @@ async def start(self):
         for preload in self.preloads:
             await preload.start()
 
+        if not os.path.exists(self.local_directory):
+            os.makedirs(self.local_directory, exist_ok=True)
```

Comment:
Do you mind co-locating this with these lines

https://github.com/dask/distributed/blob/266c98c16c855c60195e90da4cea5a7a85f69f27/distributed/nanny.py#L115-L124

so all the logic around parsing / creating `self.local_directory` is in the same place